In [1]:
import pandas as pd


In [3]:
data = pd.read_csv("Dataset .csv")
data.head()


,Restaurant ID,Restaurant Name,Country Code,City,Address,Locality,Locality Verbose,Longitude,Latitude,Cuisines,...,Currency,Has Table booking,Has Online delivery,Is delivering now,Switch to order menu,Price range,Aggregate rating,Rating color,Rating text,Votes
0,6317637,Le Petit Souffle,162,Makati City,"Third Floor, Century City Mall, Kalayaan Avenu...","Century City Mall, Poblacion, Makati City","Century City Mall, Poblacion, Makati City, Mak...",121.027535,14.565443,"French, Japanese, Desserts",...,Botswana Pula(P),Yes,No,No,No,3,4.8,Dark Green,Excellent,314
1,6304287,Izakaya Kikufuji,162,Makati City,"Little Tokyo, 2277 Chino Roces Avenue, Legaspi...","Little Tokyo, Legaspi Village, Makati City","Little Tokyo, Legaspi Village, Makati City, Ma...",121.014101,14.553708,Japanese,...,Botswana Pula(P),Yes,No,No,No,3,4.5,Dark Green,Excellent,591
2,6300002,Heat - Edsa Shangri-La,162,Mandaluyong City,"Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...","Edsa Shangri-La, Ortigas, Mandaluyong City","Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...",121.056831,14.581404,"Seafood, Asian, Filipino, Indian",...,Botswana Pula(P),Yes,No,No,No,4,4.4,Green,Very Good,270
3,6318506,Ooma,162,Mandaluyong City,"Third Floor, Mega Fashion Hall, SM Megamall, O...","SM Megamall, Ortigas, Mandaluyong City","SM Megamall, Ortigas, Mandaluyong City, Mandal...",121.056475,14.585318,"Japanese, Sushi",...,Botswana Pula(P),No,No,No,No,4,4.9,Dark Green,Excellent,365
4,6314302,Sambo Kojin,162,Mandaluyong City,"Third Floor, Mega Atrium, SM Megamall, Ortigas...","SM Megamall, Ortigas, Mandaluyong City","SM Megamall, Ortigas, Mandaluyong City, Mandal...",121.057508,14.584450,"Japanese, Korean",...,Botswana Pula(P),Yes,No,No,No,4,4.8,Dark Green,Excellent,229


In [4]:
restaurant_counts = data["Restaurant Name"].value_counts()
chains = restaurant_counts[restaurant_counts > 1]
chains.head(10)


,count
Restaurant Name,
Cafe Coffee Day,83
Domino's Pizza,79
Subway,63
Green Chick Chop,51
McDonald's,48
Keventers,34
Pizza Hut,30
Giani,29
Baskin Robbins,28


In [5]:
chains_df = data[data["Restaurant Name"].isin(chains.index)]

chain_popularity = chains_df["Restaurant Name"].value_counts().head(10)
chain_popularity


,count
Restaurant Name,
Cafe Coffee Day,83
Domino's Pizza,79
Subway,63
Green Chick Chop,51
McDonald's,48
Keventers,34
Pizza Hut,30
Giani,29
Baskin Robbins,28


In [6]:
chain_ratings = (
    chains_df
    .groupby("Restaurant Name")["Aggregate rating"]
    .mean()
    .sort_values(ascending=False)
)

chain_ratings.head(10)


,Aggregate rating
Restaurant Name,
Talaga Sampireun,4.900
Silantro Fil-Mex,4.850
AB's Absolute Barbecues,4.850
AB's - Absolute Barbecues,4.825
Naturals Ice Cream,4.800
Gymkhana,4.700
The Cheesecake Factory,4.650
Garota de Ipanema,4.600
Dishoom,4.600


In [7]:
top_chains = chain_popularity.index
top_chain_ratings = chain_ratings.loc[top_chains]
top_chain_ratings


,Aggregate rating
Restaurant Name,
Cafe Coffee Day,2.419277
Domino's Pizza,2.740506
Subway,2.907937
Green Chick Chop,2.672549
McDonald's,3.339583
Keventers,2.870588
Pizza Hut,3.320000
Giani,2.689655
Baskin Robbins,1.860714


Conclusion:
The analysis confirms the presence of several restaurant chains in the dataset, identified by multiple outlets operating under the same restaurant name. Popular chains such as Cafe Coffee Day, Domino’s Pizza, and Subway have the highest number of outlets, indicating strong brand presence and wide market reach.

When analyzing ratings, some highly popular chains show moderate average ratings. For example, Cafe Coffee Day and Domino’s Pizza have average ratings in the range of approximately 2.4–2.7, suggesting that popularity does not always correspond to higher customer satisfaction.

On the other hand, certain chains such as Barbeque Nation demonstrate both a strong presence and a high average rating (above 4.3), indicating a balance between popularity and quality. Additionally, chains like AB’s Absolute Barbecues, Naturals Ice Cream, and The Cheesecake Factory receive very high average ratings despite having fewer outlets.

Overall, the analysis highlights that while some restaurant chains are widely popular, higher customer ratings are often associated with select premium or niche chains rather than mass-market brands.